src/BotwinExtensions.cs

```diff
@@ -115,42 +120,68 @@ private static RequestDelegate CreateRouteHandler(string path, Type moduleType)
             };
         }
 
-        private static void ApplyGlobalAfterHook(IApplicationBuilder builder, BotwinOptions options)
+        private static void ApplyGlobalAfterHook(IApplicationBuilder builder, BotwinOptions options, ILogger logger)
         {
             if (options?.After != null)
             {
+                logger?.LogTrace("Applying global after hook");
                 builder.Use(async (ctx, next) =>
                 {
                     await next();
+                    logger?.LogTrace("Executing global after hook");
```

Comment:
This again is inside of a middleware so it should probably use a different logger.